tests/avifgridapitest.c

```diff
@@ -29,7 +29,8 @@ static avifBool createImage(int width, int height, int depth, avifPixelFormat yu
     uint32_t uvWidth = ((*image)->width + formatInfo.chromaShiftX) >> formatInfo.chromaShiftX;
     uint32_t uvHeight = ((*image)->height + formatInfo.chromaShiftY) >> formatInfo.chromaShiftY;
 
-    for (uint32_t plane = 0; plane < (formatInfo.monochrome ? 1 : AVIF_PLANE_COUNT_YUV); ++plane) {
+    const int planeCount = formatInfo.monochrome ? 1 : AVIF_PLANE_COUNT_YUV;
```

Comment:
Is there any guideline for picking `int` over `uint32_t` besides avoiding warnings? I found a mix of types in the code base. Example: `avifImageCreate(int width, int height)` but `uint32_t avifImage::width`.

Reply:
In general I would suggest going for consistency.

For the type of the loop index variable `plane`, Joe and I looked at existing code in libavif that iterates over the planes (I searched for "< 3;" and "AVIF_PLANE_COUNT_YUV"). We found that the` int` type is commonly used for this index variable. That's why we went with `int` for consistency.

`avifImageCreate(int width, int height)` vs `uint32_t avifImage::width` does seem like unintended inconsistency, because the three `int` parameters are assigned to `uint32_t` struct members without negative value checks:

```
avifImage * avifImageCreate(int width, int height, int depth, avifPixelFormat yuvFormat)
{
    avifImage * image = (avifImage *)avifAlloc(sizeof(avifImage));
    avifImageSetDefaults(image);
    image->width = width;
    image->height = height;
    image->depth = depth;
    image->yuvFormat = yuvFormat;
    return image;
}
```